diff --git a/media_kit/lib/src/player/native/core/initializer.dart b/media_kit/lib/src/player/native/core/initializer.dart
index 1963c5c3..14252451 100644
--- a/media_kit/lib/src/player/native/core/initializer.dart
+++ b/media_kit/lib/src/player/native/core/initializer.dart
@@ -7,0 +8 @@ import 'dart:ffi';
+import 'dart:io';
@@ -27,0 +29,7 @@ abstract class Initializer {
+    if (Platform.isAndroid) {
+      return InitializerIsolate.create(
+        mpv,
+        callback,
+        options,
+      );
+    }
@@ -46,0 +55,4 @@ abstract class Initializer {
+    if (Platform.isAndroid) {
+      InitializerIsolate.dispose(handle);
+      return;
+    }
