diff --git a/libs/android/media_kit_libs_android_video/android/build.gradle b/libs/android/media_kit_libs_android_video/android/build.gradle
index 5d6f366..b386b5a 100644
--- a/libs/android/media_kit_libs_android_video/android/build.gradle
+++ b/libs/android/media_kit_libs_android_video/android/build.gradle
@@ -53,36 +53,22 @@ task downloadDependencies(type: Exec) {
         }
         outputDir.mkdirs()

-        // Download all *.jar dependencies & verify their MD5 checksums.
+        // Download all *.jar dependencies.
         def filesToDownload = [
-            ["url": "https://github.com/bggRGjQaUbCoE/libmpv-android-video-build/releases/download/vnext/default-arm64-v8a.jar", "md5": "2451ce9f388c67cf19e26e1f8f4caf0c", "destination": file("$buildDir/v1.1.6/default-arm64-v8a.jar")],
-            ["url": "https://github.com/bggRGjQaUbCoE/libmpv-android-video-build/releases/download/vnext/default-armeabi-v7a.jar", "md5": "9d6caaa4d6bf660f4d9a611c4deab5af", "destination": file("$buildDir/v1.1.6/default-armeabi-v7a.jar")],
-            ["url": "https://github.com/bggRGjQaUbCoE/libmpv-android-video-build/releases/download/vnext/default-x86_64.jar", "md5": "e54e49b7c87d2367c515d7c24c934fbb", "destination": file("$buildDir/v1.1.6/default-x86_64.jar")],
+            ["url": "https://github.com/bggRGjQaUbCoE/libmpv-android-video-build/releases/download/vnext/default-arm64-v8a.jar", "destination": file("$buildDir/v1.1.6/default-arm64-v8a.jar")],
+            ["url": "https://github.com/bggRGjQaUbCoE/libmpv-android-video-build/releases/download/vnext/default-armeabi-v7a.jar", "destination": file("$buildDir/v1.1.6/default-armeabi-v7a.jar")],
+            ["url": "https://github.com/bggRGjQaUbCoE/libmpv-android-video-build/releases/download/vnext/default-x86_64.jar", "destination": file("$buildDir/v1.1.6/default-x86_64.jar")],
         ]

         filesToDownload.each { fileInfo ->
             def destFile = fileInfo.destination
             if (destFile.exists()) {
-                def calculatedMD5 = MessageDigest.getInstance("MD5").digest(Files.readAllBytes(destFile.toPath())).encodeHex().toString()
-
-                if (calculatedMD5 != fileInfo.md5) {
-                    destFile.delete()
-                    println "MD5 mismatch. File deleted: ${destFile}"
-                }
-
-            }
-
-            if (!destFile.exists()) {
+            } else {
                 destFile.parentFile.mkdirs()
                 println "Downloading file from: ${fileInfo.url}"
                 destFile.withOutputStream { os ->
                     os << new URL(fileInfo.url).openStream()
                 }
-
-                def calculatedMD5 = MessageDigest.getInstance("MD5").digest(Files.readAllBytes(destFile.toPath())).encodeHex().toString()
-                if (calculatedMD5 != fileInfo.md5) {
-                    throw new GradleException("MD5 verification failed for ${destFile}")
-                }
             }

             copy {
                 from destFile
                 into outputDir
             }
         }
 }

 assemble.dependsOn(downloadDependencies)
\ No newline at end of file